<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Qirui Zhang" />

<meta name="date" content="2025-06-26" />

<title>ApoE_pheno</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">diffdriver_run</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/LD-ZhangQR/diffdriver">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">ApoE_pheno</h1>
<h4 class="author">Qirui Zhang</h4>
<h4 class="date">2025-06-26</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-06-26
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>diffdriver/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250319code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250319)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250319code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250319)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomLDZhangQRdiffdriveranalysistreecb90e40de7e2ea3c76dcb229b48398ddc0781aa4targetblankcb90e40a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/LD-ZhangQR/diffdriver-analysis/tree/cb90e40de7e2ea3c76dcb229b48398ddc0781aa4" target="_blank">cb90e40</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomLDZhangQRdiffdriveranalysistreecb90e40de7e2ea3c76dcb229b48398ddc0781aa4targetblankcb90e40a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/LD-ZhangQR/diffdriver-analysis/tree/cb90e40de7e2ea3c76dcb229b48398ddc0781aa4" target="_blank">cb90e40</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  code/run_workflowr.R

Unstaged changes:
    Deleted:    analysis/clinical_newplot.Rmd
    Deleted:    analysis/example.Rmd
    Modified:   analysis/immunesubtypes.Rmd
    Deleted:    analysis/immunesubtypes_newplot.Rmd
    Modified:   analysis/index.Rmd
    Modified:   diffdriver.Rproj

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/ApoE_pheno.Rmd</code>) and HTML
(<code>docs/ApoE_pheno.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LD-ZhangQR/diffdriver-analysis/blob/cb90e40de7e2ea3c76dcb229b48398ddc0781aa4/analysis/ApoE_pheno.Rmd" target="_blank">cb90e40</a>
</td>
<td>
Qirui Zhang
</td>
<td>
2025-06-26
</td>
<td>
Add ApoE_pheno
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>This document explains how the shell script calls GDC data, extracts
the APOE genomic window, performs SNP calling, applies quality‑control
filters, and translates the two APOE SNPs (rs429358 &amp; rs7412) into
the classical ε2/ε3/ε4 genotypes.</p>
<pre class="r"><code>#!/bin/bash -l
#SBATCH --job-name=apoe_call
#SBATCH --output=/dartfs/rc/lab/S/Szhao/qiruiz/GDC/new_result/apoe_%A_%a.out
#SBATCH --time=24:00:00
#SBATCH --partition=standard
#SBATCH --account=nccc
#SBATCH --nodes=1
#SBATCH --ntasks=1
#SBATCH --cpus-per-task=4
#SBATCH --mem=32G
#SBATCH --mail-type=BEGIN,END,FAIL
#SBATCH --array=1-2000</code></pre>
<p>These variables define where the pipeline will look for input lists,
authentication credentials, and the reference genome.</p>
<pre class="r"><code>set -euo pipefail
echo &quot;[Task $SLURM_ARRAY_TASK_ID]  $SLURM_JOB_ID   $(hostname)  $(date)&quot;
source /optnfs/common/miniconda3/etc/profile.d/conda.sh
conda activate ega    
cd /dartfs/rc/lab/S/Szhao/qiruiz/GDC/gdc-client-v1.6.1          # gdc-client
manifest=&quot;gdc_manifest.2025-05-01.153233.txt&quot;
sample_sheet=&quot;gdc_sample_sheet.2025-05-01.tsv&quot;
token_file=&quot;gdc-user-token.2025-06-10T15_22_33.866Z.txt&quot;
reference_genome=&quot;/dartfs/rc/lab/S/Szhao/qiruiz/GDC/GRCh38.primary_assembly.genome.fa&quot;</code></pre>
<pre class="r"><code>chrom=19;  start=44908184;  end=44909322    # fully covering APOE (GRCh38)
min_dp=3              # minimum read depth per SNP
threads=4            # CPUs used by samtools/bcftools
out_root=&quot;/dartfs/rc/lab/S/Szhao/qiruiz/GDC/new_apoe_analysis_output&quot;
mkdir -p &quot;$out_root&quot;

master_tsv=&quot;$out_root/ApoE_genotypes.tsv&quot;     # Success table 
failed_tsv=&quot;$out_root/failed_samples.tsv&quot;     # Failure table

# Initialise result tables once
if [[ ! -s $master_tsv ]]; then
  printf &quot;Case_ID\tSample_ID\tFile_ID\tArray_Index\tGT_rs429358\tAlleles_rs429358\tDP_rs429358\tGT_rs7412\tAlleles_rs7412\tDP_rs7412\tAPOE_genotype\n&quot; &gt; &quot;$master_tsv&quot;
fi
if [[ ! -s $failed_tsv ]]; then
  printf &quot;Case_ID\tSample_ID\tFile_ID\tArray_Index\tReason\n&quot; &gt; &quot;$failed_tsv&quot;
fi</code></pre>
<p>Maps the SLURM array index to the right manifest row, fetches
Case_ID/Sample_ID, and creates a unique working directory.</p>
<pre class="r"><code>idx=$SLURM_ARRAY_TASK_ID
line=$((idx + 1))  # +1 to skip manifest header

# Read file_id &amp; filename from manifest
IFS=$&#39;\t&#39; read -r file_id filename _md5 _size _state &lt; &lt;(sed -n &quot;${line}p&quot; &quot;$manifest&quot;)

# Look up Case_ID and Sample_ID in sample_sheet
case_id=$(awk -F&#39;\t&#39; -v id=&quot;$file_id&quot; &#39;NR&gt;1 &amp;&amp; $1==id {print $6}&#39; &quot;$sample_sheet&quot;)
sample_id=$(awk -F&#39;\t&#39; -v id=&quot;$file_id&quot; &#39;NR&gt;1 &amp;&amp; $1==id {print $7}&#39; &quot;$sample_sheet&quot;)
echo &quot;[Task $idx]  File_ID=$file_id   Case=$case_id   Sample=$sample_id&quot;

# Unique per-sample working directory
work_dir=&quot;${out_root}/${sample_id}_${file_id}&quot;
mkdir -p &quot;$work_dir&quot;</code></pre>
<div id="download-bam" class="section level1">
<h1>Download BAM</h1>
<p>Uses a one-row manifest for gdc-client, retries up to five times, and
logs any permanent failure.</p>
<pre class="r"><code>tmp_manifest=$(mktemp)
{ head -n1 &quot;$manifest&quot;; sed -n &quot;${line}p&quot; &quot;$manifest&quot;; } &gt; &quot;$tmp_manifest&quot;

tries=0; ok=0
while ((tries &lt; 5)); do
  ./gdc-client download -m &quot;$tmp_manifest&quot; -t &quot;$token_file&quot; &amp;&amp; { ok=1; break; }
  echo &quot;[Task $idx]  download attempt $((tries+1)) failed – retrying…&quot; &gt;&amp;2
  sleep 60; ((tries++))
done
rm -f &quot;$tmp_manifest&quot;

if ((ok == 0)); then
  printf &quot;%s\t%s\t%s\t%d\tDownload_failed\n&quot; &quot;$case_id&quot; &quot;$sample_id&quot; &quot;$file_id&quot; &quot;$idx&quot; &gt;&gt; &quot;$failed_tsv&quot;
  exit 0
fi</code></pre>
<pre class="r"><code># Move downloaded BAM (and .bai) into work_dir
mv &quot;${file_id}/$filename&quot; &quot;$work_dir/&quot;
[[ -f &quot;${file_id}/${filename%.bam}.bai&quot; ]] &amp;&amp; mv &quot;${file_id}/${filename%.bam}.bai&quot; &quot;$work_dir/&quot;
rm -rf &quot;$file_id&quot;
bam=&quot;$work_dir/$filename&quot;

# Build index if missing
[[ ! -f &quot;$bam.bai&quot; &amp;&amp; ! -f &quot;${bam%.bam}.bai&quot; ]] &amp;&amp; samtools index -@ &quot;$threads&quot; &quot;$bam&quot;</code></pre>
<pre class="r"><code>if samtools idxstats &quot;$bam&quot; | awk &#39;NR==1{print $1}&#39; | grep -q &#39;^chr&#39;; then
  pfix=&quot;chr&quot;
else
  pfix=&quot;&quot;
fi

region=&quot;${pfix}${chrom}:${start}-${end}&quot;
snp429358=&quot;${pfix}${chrom}:44908684&quot;   # rs429358
snp7412=&quot;${pfix}${chrom}:44908822&quot;     # rs7412
echo &quot;[DEBUG] prefix=$pfix  region=$region&quot;

region_bam=&quot;$work_dir/region.bam&quot;
samtools view -@ &quot;$threads&quot; -bh &quot;$bam&quot; &quot;$region&quot; -o &quot;$region_bam&quot;
samtools index -@ &quot;$threads&quot; &quot;$region_bam&quot;
rm -f &quot;$bam&quot; &quot;${bam%.bam}.bai&quot;         # Remove full BAM to save disk</code></pre>
</div>
<div id="variant-calling-built-in-qc" class="section level1">
<h1>Variant Calling &amp; Built-In QC</h1>
<p>Performs bcftools mpileup and bcftools call, filtering on
mapping‐quality (MQ ≥30) and base‐quality (BQ ≥20), and preserving
per‐site depth (FORMAT/DP) for downstream checks.</p>
<p>-q 30 enforces a mapping quality ≥30 (drops reads with ambiguous
alignments)</p>
<p>-Q 20 enforces a base quality ≥20 (drops likely‐erroneous bases)</p>
<pre class="r"><code>vcf=&quot;$work_dir/variants.vcf.gz&quot;
echo &quot;[Task $idx] Starting variant calling at $(date)&quot;
bcftools mpileup  -f &quot;$reference_genome&quot; \
        -q 30 -Q 20 -a FORMAT/DP \
        --threads &quot;$threads&quot; &quot;$region_bam&quot; -Ou | \
bcftools call -m -A --threads &quot;$threads&quot; -Oz -o &quot;$vcf&quot;

bcftools index &quot;$vcf&quot;

echo &quot;[Task $idx] Variant calling completed at $(date)&quot;</code></pre>
</div>
<div id="extract-gtdp" class="section level1">
<h1>Extract GT/DP</h1>
<p>Use bcftools query to extract REF, ALT, genotype (GT), and depth (DP)
for the two SNPs (rs429358 &amp; rs7412). Check for missing genotype
calls (./.) or depth below min_dp, appending any failures to the fail
string. If any QC checks fail, write the failure reason to failed_tsv
and exit early.</p>
<pre class="r"><code>query_fmt=&#39;%REF %ALT [%GT] [%DP]\n&#39;       

read -r ref1 alt1 gt1 dp1 &lt; &lt;(bcftools query -r &quot;$snp429358&quot; -f &quot;$query_fmt&quot; &quot;$vcf&quot;) || true
read -r ref2 alt2 gt2 dp2 &lt; &lt;(bcftools query -r &quot;$snp7412&quot;   -f &quot;$query_fmt&quot; &quot;$vcf&quot;) || true
fail=&quot;&quot;
for tag in 1 2; do
  gt_var=&quot;gt${tag}&quot;; dp_var=&quot;dp${tag}&quot;
  gt_val=${!gt_var}; dp_val=${!dp_var}
  snp_name=$([[ $tag == 1 ]] &amp;&amp; echo rs429358 || echo rs7412)

  [[ -z $gt_val || $gt_val == &quot;./.&quot; ]] &amp;&amp; fail+=&quot;NoCall_${snp_name};&quot;
  [[ -z $dp_val || $dp_val == &quot;.&quot; || $dp_val -lt $min_dp ]] &amp;&amp; \
      fail+=&quot;LowDP_${snp_name}(${dp_val:-0});&quot;
done

if [[ -n $fail ]]; then
    echo -e &quot;$case_id\t$sample_id\t$file_id\t$idx\t$fail&quot; &gt;&gt; &quot;$failed_tsv&quot;
    exit 0
fi</code></pre>
</div>
<div id="allele-translation-ε-genotype-mapping" class="section level1">
<h1>Allele Translation &amp; ε-Genotype Mapping</h1>
<p>allele() function converts a numeric allele index (0,1) into its
actual nucleotide or falls back to the reference if missing. Split GT
strings (0/1, 1|1, etc.) into two haplotypes for each SNP. Resolve bases
for both alleles at rs429358 and rs7412, then join them into allele1 and
allele2. Count C’s at rs429358 (ε4 allele) and T’s at rs7412 (ε2 allele)
to get counts c4 and t2. Map counts to the six possible ε genotypes via
a case statement; any unexpected combo is logged as a QC failure. Format
the output line with all fields, print it for debugging, append it to
the master TSV, and log completion.</p>
<pre class="r"><code>allele() {
    local idx=$1 ref=$2 alt=$3
    [[ $idx == 0 ]] &amp;&amp; echo &quot;$ref&quot; || { [[ $alt == &quot;.&quot; ]] &amp;&amp; echo &quot;$ref&quot; || echo &quot;$alt&quot;; }
}
# split &quot;0/1&quot; etc.
IFS=&#39;|/&#39; read -r gt1a gt1b &lt;&lt;&lt;&quot;$gt1&quot;   
IFS=&#39;|/&#39; read -r gt2a gt2b &lt;&lt;&lt;&quot;$gt2&quot; 

r1a=$(allele &quot;$gt1a&quot; &quot;$ref1&quot; &quot;$alt1&quot;); r1b=$(allele &quot;$gt1b&quot; &quot;$ref1&quot; &quot;$alt1&quot;)
r2a=$(allele &quot;$gt2a&quot; &quot;$ref2&quot; &quot;$alt2&quot;); r2b=$(allele &quot;$gt2b&quot; &quot;$ref2&quot; &quot;$alt2&quot;)

allele1=&quot;${r1a}/${r1b}&quot;
allele2=&quot;${r2a}/${r2b}&quot;

#   ε4  ↔  C at rs429358
#   ε2  ↔  T at rs7412
c4=0
[[ $r1a == &quot;C&quot; ]] &amp;&amp; ((++c4))
[[ $r1b == &quot;C&quot; ]] &amp;&amp; ((++c4))

t2=0
[[ $r2a == &quot;T&quot; ]] &amp;&amp; ((++t2))
[[ $r2b == &quot;T&quot; ]] &amp;&amp; ((++t2))

case &quot;${c4}${t2}&quot; in
    20) genotype=&quot;ε4/ε4&quot; ;;
    10) genotype=&quot;ε3/ε4&quot; ;;
    11) genotype=&quot;ε2/ε4&quot; ;;
    00) genotype=&quot;ε3/ε3&quot; ;;
    01) genotype=&quot;ε2/ε3&quot; ;;
    02) genotype=&quot;ε2/ε2&quot; ;;
    *)  genotype=&quot;?&quot; ;
        echo -e &quot;$case_id\t$sample_id\t$file_id\t$idx\tUnexpected_combination&quot; &gt;&gt; &quot;$failed_tsv&quot; ;;
esac

line=$(printf &quot;%s\t%s\t%s\t%d\t%s\t%s\t%d\t%s\t%s\t%d\t%s&quot; \
              &quot;$case_id&quot; &quot;$sample_id&quot; &quot;$file_id&quot; &quot;$idx&quot; \
              &quot;$gt1&quot; &quot;$allele1&quot; &quot;$dp1&quot; \
              &quot;$gt2&quot; &quot;$allele2&quot; &quot;$dp2&quot; \
              &quot;$genotype&quot;)

echo &quot;[DBG] printf RC=$?  line=&#39;$line&#39;&quot;

echo -e &quot;$line&quot;                         

echo -e &quot;$line&quot; &gt;&gt; &quot;$master_tsv&quot;

echo &quot;[Task $idx] finished  $(date)&quot;</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.2 (2024-10-31)
Platform: x86_64-conda-linux-gnu
Running under: Ubuntu 24.04.1 LTS

Matrix products: default
BLAS/LAPACK: /dartfs-hpc/rc/home/p/f0070pp/.conda/envs/diffdriver/lib/libopenblasp-r0.3.29.so;  LAPACK version 3.12.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: Etc/UTC
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.7.1

loaded via a namespace (and not attached):
 [1] vctrs_0.6.5       httr_1.4.7        cli_3.6.5         knitr_1.50       
 [5] rlang_1.1.6       xfun_0.52         stringi_1.8.7     processx_3.8.6   
 [9] promises_1.3.3    jsonlite_2.0.0    glue_1.8.0        rprojroot_2.0.4  
[13] git2r_0.36.2      htmltools_0.5.8.1 httpuv_1.6.16     ps_1.9.1         
[17] sass_0.4.10       rmarkdown_2.29    jquerylib_0.1.4   tibble_3.2.1     
[21] evaluate_1.0.3    fastmap_1.2.0     yaml_2.3.10       lifecycle_1.0.4  
[25] whisker_0.4.1     stringr_1.5.1     compiler_4.4.2    fs_1.6.6         
[29] pkgconfig_2.0.3   Rcpp_1.0.14       rstudioapi_0.17.1 later_1.4.2      
[33] digest_0.6.37     R6_2.6.1          pillar_1.10.2     callr_3.7.6      
[37] magrittr_2.0.3    bslib_0.9.0       tools_4.4.2       cachem_1.1.0     
[41] getPass_0.2-4    </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
