<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Qirui Zhang" />

<meta name="date" content="2025-03-23" />

<title>example</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="site_libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.33/datatables.js"></script>
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">diffdriver_run</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/LD-ZhangQR/diffdriver">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">example</h1>
<h4 class="author">Qirui Zhang</h4>
<h4 class="date">2025-03-23</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-04-08
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>diffdriver/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250319code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250319)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250319code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250319)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/dartfs/rc/lab/S/Szhao/qiruiz/diffdriver/workflowr/diffdriver/output/
</td>
<td style="text-align:left;">
output
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomLDZhangQRdiffdriveranalysistreee61ca180f1705895533fb1dd82b2b36dc926a667targetblanke61ca18a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/LD-ZhangQR/diffdriver-analysis/tree/e61ca180f1705895533fb1dd82b2b36dc926a667" target="_blank">e61ca18</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomLDZhangQRdiffdriveranalysistreee61ca180f1705895533fb1dd82b2b36dc926a667targetblanke61ca18a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/LD-ZhangQR/diffdriver-analysis/tree/e61ca180f1705895533fb1dd82b2b36dc926a667" target="_blank">e61ca18</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  code/run_workflowr.R
    Untracked:  output/BLCA_FGFR3_Aneuploidy.Score_reg59e6f336fd3d8

Unstaged changes:
    Deleted:    analysis/example.Rmd
    Modified:   diffdriver.Rproj

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/PRS.Rmd</code>) and HTML
(<code>docs/PRS.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LD-ZhangQR/diffdriver-analysis/65df6097c11c1e62f4317b077bffa94e99c1f7c6/docs/PRS.html" target="_blank">65df609</a>
</td>
<td>
Qirui Zhang
</td>
<td>
2025-04-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LD-ZhangQR/diffdriver-analysis/blob/442ada3a1d1396c8cee8409c5d93f3aa6b2314c1/analysis/PRS.Rmd" target="_blank">442ada3</a>
</td>
<td>
Qirui Zhang
</td>
<td>
2025-04-03
</td>
<td>
Update all analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>knitr::opts_chunk$set(echo = TRUE)</code></pre>
<pre class="r"><code>get_files &lt;- function(outputdir, tumors = NULL, phenotypes = NULL) {
  files &lt;- list.files(
    path = outputdir,
    pattern = &quot;_resdd\\.Rd$&quot;,
    recursive = TRUE,
    full.names = TRUE
  )
  
  matching_files_txt &lt;- sub(&quot;\\.Rd$&quot;, &quot;.txt&quot;, files)
  
  analysis_mode &lt;- ifelse(grepl(&quot;_sig_&quot;, basename(files)), &quot;sig&quot;, &quot;reg&quot;)

  inner_folder_names      &lt;- basename(dirname(files))            # e.g. &quot;Aneuploidy.Score_BLCA_mutations&quot;
  phenotype_folder_names  &lt;- basename(dirname(dirname(files)))   # e.g. &quot;Aneuploidy.Score&quot;
  
  tumor_names &lt;- mapply(function(inner_name, phen) {
    pattern_front &lt;- paste0(&quot;^&quot;, phen, &quot;_&quot;)
    tmp &lt;- sub(pattern_front, &quot;&quot;, inner_name)
    tmp &lt;- sub(&quot;_mutations$&quot;, &quot;&quot;, tmp)
    return(tmp)
  }, inner_folder_names, phenotype_folder_names)
  
  phenos_all &lt;- phenotype_folder_names
  
  filedf &lt;- data.frame(
    tumor      = tumor_names,
    phenotype  = phenos_all,
    mode       = analysis_mode,   
    file       = files,
    filetxt    = matching_files_txt,
    stringsAsFactors = FALSE
  )
  
  if (!is.null(tumors)) {
    filedf &lt;- filedf[filedf$tumor %in% tumors, ]
  }
  if (!is.null(phenotypes)) {
    filedf &lt;- filedf[filedf$phenotype %in% phenotypes, ]
  }

  rownames(filedf) &lt;- NULL
  return(filedf)
}

filedf &lt;- get_files(
  outputdir = &quot;/dartfs/rc/lab/S/Szhao/qiruiz/diffdriver/temp/output/PRS&quot;
)

head(filedf)</code></pre>
<pre><code>  tumor        phenotype mode
1  BLCA Allergic_disease  reg
2  BLCA Allergic_disease  sig
3  BRCA Allergic_disease  reg
4  BRCA Allergic_disease  sig
5  CESC Allergic_disease  reg
6  CESC Allergic_disease  sig
                                                                                                                                                    file
1 /dartfs/rc/lab/S/Szhao/qiruiz/diffdriver/temp/output/PRS/Allergic_disease/Allergic_disease_BLCA_mutations/testdiffdriver_reg_Allergic_disease_resdd.Rd
2 /dartfs/rc/lab/S/Szhao/qiruiz/diffdriver/temp/output/PRS/Allergic_disease/Allergic_disease_BLCA_mutations/testdiffdriver_sig_Allergic_disease_resdd.Rd
3 /dartfs/rc/lab/S/Szhao/qiruiz/diffdriver/temp/output/PRS/Allergic_disease/Allergic_disease_BRCA_mutations/testdiffdriver_reg_Allergic_disease_resdd.Rd
4 /dartfs/rc/lab/S/Szhao/qiruiz/diffdriver/temp/output/PRS/Allergic_disease/Allergic_disease_BRCA_mutations/testdiffdriver_sig_Allergic_disease_resdd.Rd
5 /dartfs/rc/lab/S/Szhao/qiruiz/diffdriver/temp/output/PRS/Allergic_disease/Allergic_disease_CESC_mutations/testdiffdriver_reg_Allergic_disease_resdd.Rd
6 /dartfs/rc/lab/S/Szhao/qiruiz/diffdriver/temp/output/PRS/Allergic_disease/Allergic_disease_CESC_mutations/testdiffdriver_sig_Allergic_disease_resdd.Rd
                                                                                                                                                  filetxt
1 /dartfs/rc/lab/S/Szhao/qiruiz/diffdriver/temp/output/PRS/Allergic_disease/Allergic_disease_BLCA_mutations/testdiffdriver_reg_Allergic_disease_resdd.txt
2 /dartfs/rc/lab/S/Szhao/qiruiz/diffdriver/temp/output/PRS/Allergic_disease/Allergic_disease_BLCA_mutations/testdiffdriver_sig_Allergic_disease_resdd.txt
3 /dartfs/rc/lab/S/Szhao/qiruiz/diffdriver/temp/output/PRS/Allergic_disease/Allergic_disease_BRCA_mutations/testdiffdriver_reg_Allergic_disease_resdd.txt
4 /dartfs/rc/lab/S/Szhao/qiruiz/diffdriver/temp/output/PRS/Allergic_disease/Allergic_disease_BRCA_mutations/testdiffdriver_sig_Allergic_disease_resdd.txt
5 /dartfs/rc/lab/S/Szhao/qiruiz/diffdriver/temp/output/PRS/Allergic_disease/Allergic_disease_CESC_mutations/testdiffdriver_reg_Allergic_disease_resdd.txt
6 /dartfs/rc/lab/S/Szhao/qiruiz/diffdriver/temp/output/PRS/Allergic_disease/Allergic_disease_CESC_mutations/testdiffdriver_sig_Allergic_disease_resdd.txt</code></pre>
<pre class="r"><code>get_diff_table &lt;- function(filedf){
  pheno_all &lt;- unique(filedf$phenotype)
  numlist &lt;- list()
  
  for (p in pheno_all){
    p_txtfiles &lt;- filedf[filedf$phenotype == p, ]
    numlist[[p]] &lt;- list()
    
    for (t in seq_len(nrow(p_txtfiles))){
      txtf  &lt;- p_txtfiles[t, &quot;filetxt&quot;]
      rdf   &lt;- p_txtfiles[t, &quot;file&quot;] 
      tumor &lt;- p_txtfiles[t, &quot;tumor&quot;] 
      mode  &lt;- p_txtfiles[t, &quot;mode&quot;]     # reg / sig
      
      env &lt;- new.env()
      load(rdf, envir = env)
      res_rdata &lt;- env$res
      
      res &lt;- read.table(txtf, header = TRUE)
      res$gene &lt;- row.names(res)
      res$mode &lt;- mode
      
      res$alpha &lt;- sapply(res$gene, function(gene){
          res_rdata[[gene]][[&quot;dd&quot;]][[&quot;res.alt&quot;]]$alpha[2]
      })
      
      sig_res &lt;- res[res$dd.fdr &lt; 0.1, ]
      if (nrow(sig_res) &gt; 0){
        numlist[[p]][[ paste0(tumor, &quot;_&quot;, mode) ]] &lt;- sig_res
      }
    }
    
    if (length(numlist[[p]]) == 0){
      numlist[[p]] &lt;- NULL
    }
  }
  
  combined_df &lt;- do.call(
    rbind,
    lapply(names(numlist), function(pheno) {
      numlist.pheno &lt;- numlist[[pheno]]
      df.pheno &lt;- do.call(rbind, lapply(names(numlist.pheno), function(tname) {
        df &lt;- numlist.pheno[[tname]]

        df$tumor &lt;- sub(&quot;_.*&quot;, &quot;&quot;, tname) 
        df$mode  &lt;- sub(&quot;.*_&quot;, &quot;&quot;, tname) 
        return(df)
      }))
      df.pheno$pheno &lt;- pheno
      return(df.pheno)
    })
  )
  return(combined_df)
}

diff_table &lt;- get_diff_table(filedf)</code></pre>
<pre class="r"><code>library(DT)
diff_table_sig &lt;- subset(diff_table, mode == &quot;sig&quot;)
datatable(diff_table_sig, options = list(pageLength = 10))</code></pre>
<div class="datatables html-widget html-fill-item" id="htmlwidget-321f8cf22c7d3f9f6363" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-321f8cf22c7d3f9f6363">{"x":{"filter":"none","vertical":false,"data":[["PTEN1","TRAF3IP21","PCDHB161","PTEN11","EP3001","SLC17A61","PTEN2","FGFR3","RGS71","SLC17A611","CUL31","FLG1","CASP8","CDKN2A","ATM1","BAP11","PCDHB1611","KRAS","TRAF3IP22","ATG16L11","CUL311","HIST1H2AG1","CTNNB11","FMN21","PIK3CA1","TGFBR21","BAP111","FRG11","VHL1","MAP3K11","DNAH51","FRG12","BAP112","FMN211","GPS21","FRG111","FOXQ1","OLFM31","SMAD41","FRG112","RHOA1","FRG113","FOXA11","SMARCA41","SMARCA411","PBRM11","SLC17A612","PTEN12","KRAS1","FRG16","THSD7A1","TP531","NFE2L21","EPHA2","CBFB1","FBXW7","TP5311"],[0.00168585897743646,0.000488726154987155,0.00485878682733078,0.00256987942914103,0.00396939991634519,0.00306307216639298,0.000244216894719887,0.0012070917975978,0.00288082520984752,0.00138306641176147,0.014249107532198,0.00206785963927174,0.000886028068932995,0.008035338764431601,0.0043666908169341,0.0101673843164822,0.0122466191115382,0.0029639166079088,0.00299992118979728,0.009083262721586299,0.00508749216244353,0.000392096455965657,0.00728117773566033,0.000122099192487865,0.000515905129105803,0.00499278380688496,0.00771146402048465,0.013875113662891,0.00834509690726068,3.87162259034114e-06,0.0165935581834516,0.0431866203653031,0.0226476792230083,0.00108643619640781,0.00216913314347299,0.0218184244790801,0.00153285484317715,0.00442387542537748,0.00406882809359167,0.00382738004054078,0.000284844072732688,0.0279869740267963,0.00302726093885193,0.0065226068132538,0.0179967274324106,0.0154758616715039,0.00562352746504495,0.00225202159080214,0.00224309938982857,0.0470442659525802,0.00163257336601774,0.03035870940805,0.00460987227119582,0.00117472485231707,0.00298029535840696,0.00194863640269114,0.00863816351858507],[0.009480431812694709,0.00187916910704073,0.00562380639133483,0.00119571866202735,0.013784216959513,0.0182994478378856,0.00117976491207916,0.00148373729502156,0.0255785449517991,0.000702404424075603,0.0156994960254714,0.000915476819707747,0.00691842222722173,0.0662935068170806,0.00654724841326617,0.009849773890491889,0.0133352194290467,0.0324210696555372,0.00761984764812726,0.0033165280562906,0.0101698348065919,9.86985289390339e-05,0.643959091848326,0.00061665989316527,0.185178972084034,0.010294065373223,0.0247155264895325,0.0291803076198577,0.478459388445093,0.00507396670534522,0.0330448940074625,0.128005227420963,0.033527032026117,0.00115270326308181,0.00369155275461751,0.0260386254555644,0.0157180907803075,0.00647392784615473,0.00543092302833247,0.00139553546574349,0.00146175125851172,0.08497380170214359,0.242182910097343,0.0153472660974998,0.0141933194216884,0.07856491427972589,0.0260599022075523,0.0191766041062386,0.00907752964947005,0.09602486872026909,0.00310699026924578,0.0756249410857557,0.00584905346404875,0.00315489653792247,0.00221828037485174,0.00272814628243086,0.008172077976220561],[0.00796323152095626,0.00446302846042165,0.00539084485966966,0.00321064559110397,0.0107725603662608,0.0320960194386829,0.000833042900283609,0.00217341765025415,0.0247588558873502,0.00103479603995181,0.0180290471432784,0.00277772065286314,0.00512706397984961,0.0879729260860778,0.00651279511632011,0.009531290006496719,0.013622919893334,0.029779564074693,0.00531357229947952,0.00296969843852933,0.0133472728663123,0.000211669838880817,0.642446217302164,0.00048771999140632,0.0929634947363069,0.0114353355428641,0.0256331357320561,0.012191628219165,0.605401603251718,0.00534431752128926,0.0364362487411359,0.123594970395301,0.0371384284515621,0.000781475883491265,0.00268668834215391,0.0194804116901591,0.0139871169013403,0.00976910648697529,0.00326580440293922,0.00164169695116618,0.00145042348347773,0.119223219681288,0.25814044512592,0.0161277546608259,0.0149186569856188,0.0803095036613307,0.0379346260508301,0.0195767286835074,0.009528161438274511,0.0442129608243491,0.000861751977136213,0.0126937339892253,0.0067570080584997,0.00152886330593197,0.00208616289542316,0.00276168664707794,0.008287492342394349],[0.0161670151652215,0.00338184782455637,0.500453473607836,0.071172916755035,0.136468545492891,0.057662178300264,0.00358893692186764,0.0900503262583178,0.0122694156197168,0.02508861375739,0.0648062411044901,0.0017891793040132,0.0108794227918297,0.0146991342870789,0.10555784239288,0.0315228997920847,0.498458189733357,0.325903568583014,0.264365579703605,0.224137931034483,0.0322550508685066,0.0286207739506133,0.561465980126695,0.00115507746902379,0.217372888696339,0.0671090177742634,0.0201765571657819,0.108695494114916,0.667444399055998,0.06447503219752609,0.227586206896552,0.68779016181274,0.0433815350389321,0.00269456330148393,0.0312574739417375,0.10426567188387,0.213901609578544,0.0038387008677736,0.01065584251415,0.126441908394598,0.284020873546915,0.10426567188387,0.0215225626446217,0.119899896033634,0.119050145250955,0.101223581757508,0.3650915385711,0.0147494984001371,0.00425566598867246,0.234915086451584,0.09710675231521119,0.177226033024544,0.118617614269788,0.00219123546831659,0.00957764517571345,0.0975061803641019,0.07051373888766441],[0.0157947364919782,0.00350438975214083,0.502799759359526,0.07197402936599689,0.14676669534309,0.0286032826921629,0.00182034514425833,0.0565523863496926,0.0118732210941953,0.0217197634640705,0.0353295128905704,0.001953125,0.00591666177419213,0.0157260444233251,0.109506388122296,0.0323153357955876,0.501133786848073,0.328827569700172,0.270789376075257,0.197530864197531,0.0308155219853063,0.0282426634731254,0.595910681859219,0.000744971491332067,0.144234878763906,0.0673276745616072,0.0110524880748716,0.06544353112516441,0.577513119554092,0.0658511980937679,0.111619163745364,0.467596936752654,0.0275496245114698,0.00177018055972546,0.0304627813715476,0.0574166740653867,0.122387581114311,0.00398324388780982,0.0100788253363646,0.129205218789413,0.176371589288749,0.0574166740653867,0.0224766405393016,0.125151751646913,0.125,0.0390184423106234,0.375,0.00735178511884685,0.00408642005623306,0.251447555760803,0.07139287272371631,0.154752048576191,0.0541790458811059,0.00216962024776778,0.009337296468258651,0.0673670097634818,0.07044483071770991],[0.0356093764068048,0.027646990920746,0.988028344316621,0.0786834537637577,0.0468548326211326,0.988910804594108,0.983412102880066,0.0689237068519557,0.978954444184203,0.988837755315,0.0825508323469696,0.989891971836624,0.009372584912711,0.990577120633974,0.118073669175601,0.986519264221453,0.987985470954869,0.271757350049433,0.0609899433655482,0.994874206248387,0.07221163036316219,0.06718693204165641,0.419475607533723,0.0034203650883374,0.101883110414749,0.987751438460023,0.0195668612611154,0.10793579022121,0.738660335798048,0.120954958614628,0.993746331929972,0.166915479730878,0.993621741820586,0.00597721744142133,0.111355381735404,0.09233781705187299,0.133641757365842,0.98347578387919,0.0422292505386247,0.145042151726466,0.167566614601912,0.09233781705187311,0.0257681484000709,0.13996696143815,0.134649520185605,0.995304032941256,0.332154861499259,0.977339654199706,0.989313329335114,0.237787266109414,0.0333482859776124,0.111807399549558,0.990386553211336,0.983528845950959,0.0299563341915053,0.0601385453270761,0.98315839814772],[0.0556333462554032,0.0307897477641908,0.0631642287553001,0.06424698572852559,0.06747979857786821,0.0306307216639298,0.008059157525756269,0.0760467832486616,0.07202063024618791,0.0138306641176147,0.0569964301287921,0.0351536138676196,0.0292389262747888,0.080353387644316,0.043666908169341,0.0796030242249981,0.0796030242249981,0.0944975174786143,0.0944975174786143,0.0363330508863452,0.0203499686497741,0.0247020767258364,0.0946553105635843,0.00769224912673549,0.00877038719479865,0.0499278380688496,0.0771146402048465,0.027750227325782,0.0834509690726068,9.67905647585284e-05,0.06637423273380651,0.0863732407306062,0.0905907168920331,0.0683276940193993,0.0683276940193993,0.0436368489581603,0.0965698551201607,0.0729939445187285,0.0729939445187285,0.00765476008108156,0.0093998544001787,0.0559739480535926,0.07568152347129831,0.0260904272530152,0.0719869097296426,0.0619034466860156,0.0562352746504495,0.0225202159080214,0.0381326896270856,0.09408853190516051,0.0538749210785856,0.0607174188161001,0.0460987227119582,0.0387659201264632,0.07450738396017401,0.06430500128880751,0.08638163518585069],[0.312854249818925,0.118387653743566,0.0731094830873527,0.0298929665506837,0.234331688311721,0.182994478378856,0.0389322420986124,0.09347544958635851,0.211411901641226,0.00702404424075603,0.0627979841018856,0.00778155296751585,0.228307933498317,0.240666471621732,0.0327362420663309,0.08667892628880321,0.08667892628880321,0.510631847074711,0.261921169810317,0.0132661122251624,0.0406793392263676,0.00621800732315914,0.901749954393567,0.038849573269412,0.44972036077551,0.10294065373223,0.247155264895325,0.0583606152397155,0.830729660147508,0.12684916763363,0.13217957602985,0.256010454841925,0.134108128104468,0.072620305574154,0.116283911770451,0.0520772509111288,0.824686919015327,0.0712132063077021,0.0712132063077021,0.00279107093148698,0.0482377915308866,0.169947603404287,0.713039957397685,0.061389064389999,0.0567732776867534,0.314259657118904,0.260599022075523,0.10386984444119,0.154318004040991,0.192049737440538,0.102530678885111,0.151249882171511,0.0584905346404875,0.104111585751441,0.0554570093712934,0.0900288273202183,0.0817207797622056],[0.262786640191556,0.281170793006564,0.0700809831757056,0.08026613977759919,0.183133526226434,0.320960194386829,0.0274904157093591,0.136925311966012,0.199392380986264,0.0103479603995181,0.0721161885731134,0.0236106255493367,0.169193111335037,0.293243086953593,0.0325639755816005,0.0885489793066709,0.0885489793066709,0.469028134176415,0.196959790571068,0.0118787937541173,0.0533890914652491,0.0133351998494915,0.901477215807779,0.0307263594585982,0.316075882103443,0.114353355428641,0.256331357320561,0.02438325643833,0.839315605943044,0.133607938032231,0.145744994964544,0.247189940790602,0.148553713806248,0.0492329806599497,0.0846306827778481,0.0389608233803181,0.800907716520648,0.107460171356728,0.107460171356728,0.00328339390233237,0.0478639749547652,0.238446439362575,0.680995438961347,0.06451101864330371,0.0596746279424752,0.321238014645323,0.379346260508301,0.154914031066674,0.161978744450667,0.0884259216486981,0.028437815245495,0.0253874679784506,0.067570080584997,0.0504524890957551,0.0521540723855791,0.09113565935357219,0.08287492342394349],[0.505168555110707,0.07101880431568371,1,0.444830729718969,0.773321757793049,0.57662178300264,0.118434918421632,0.577340927452463,0.306735390492919,0.155181561831178,0.259224964417961,0.0304160481682243,0.35902095213038,0.146991342870789,0.6630601021439519,0.359370638695053,0.877943446802507,1,1,0.620721436516765,0.129020203474026,0.42370333471841,0.856917961555913,0.07276988054849889,0.739067821567552,0.620317573401351,0.201765571657819,0.217390988229833,1,0.439827873914768,0.9103448275862061,0.68779016181274,0.173526140155729,0.169757487993487,0.218802317592162,0.20853134376774,0.942799597347051,0.126677128636529,0.175821401483475,0.180760032063965,0.935676699201717,0.20853134376774,0.538064066115543,0.479599584134538,0.476200581003821,0.404894327030033,0.91272884642775,0.147494984001371,0.07234632180743179,0.277740911821951,1,0.354452066049087,0.871047531068365,0.0723107704544476,0.239441129392836,0.558444299672727,0.705137388876644],[0.496935483387718,0.07359218479495749,1,0.449837683537481,0.818205299960959,0.286032826921629,0.060071389760525,0.508971477147234,0.296830527354882,0.144995789303663,0.141318051562282,0.033203125,0.19524983854834,0.157260444233251,0.538602404388952,0.334222880384202,0.878270885073606,0.9579373714334,0.9579373714334,0.6342021033379061,0.123262087941225,0.311718879632863,0.863482879650901,0.0469332039539202,0.490398587797279,0.616826086282649,0.110524880748716,0.130887062250329,1,0.445733265194503,0.446476654981456,0.467596936752654,0.110198498045879,0.111521375262704,0.19191552264075,0.114833348130773,0.877054924022883,0.131447048297724,0.166300618050017,0.129205218789413,0.889433273626956,0.114833348130773,0.382242801416714,0.500607006587652,0.5,0.156073769242493,0.890409338093152,0.0735178511884685,0.06946914095596191,0.251447555760803,1,0.309504097152382,0.541790458811059,0.07159746817633671,0.233432411706466,0.553743674241941,0.7044483071770991],[0.636178931435901,0.43544010700175,0.988028344316621,0.491771586023486,0.51224732519535,0.988910804594108,0.983412102880066,0.720200009240463,0.978954444184203,0.988837755315,0.330203329387878,1,0.309295302119463,0.990577120633974,0.594236493120462,0.988880913437802,0.988880913437802,0.9582413858628021,0.9582413858628021,0.994874206248387,0.288846521452649,0.705462786437392,0.81404824697231,0.215483000565256,0.988958787226045,0.990532016217221,0.195668612611154,0.21587158044242,0.83900700932446,0.755968491341423,0.993746331929972,0.333830959461756,0.993922218955287,0.376564698809544,0.690383244416802,0.184675634103746,0.8347705153382,0.98347578387919,0.574340626340533,0.145042151726466,0.9217993423538851,0.184675634103746,0.644203710001772,0.559867845752599,0.538598080742422,0.995304032941256,0.664309722998518,0.977339654199706,0.993700990295169,0.237787266109414,0.821078609788877,0.223614799099116,0.990386553211336,0.983528845950959,0.643492672691279,0.500076839273832,0.98315839814772],[8,12,2,1,3,5,0,25,0,0,7,10,28,7,8,0,0,6,4,0,7,1,13,4,11,5,2,1,59,14,3,5,0,19,7,6,1,0,9,6,6,1,2,1,1,0,1,0,9,5,17,5,0,10,9,13,0],[1,1,0,7,9,0,9,15,6,5,1,0,11,0,2,6,2,3,9,2,1,7,19,20,19,0,11,6,56,5,0,2,4,5,1,1,5,9,1,1,2,6,11,6,6,4,4,7,0,2,8,13,4,0,1,6,5],[181,144,79,333,55,56,194,132,362,61,91,56,190,58,175,70,70,136,136,15,88,152,69,144,57,62,178,27,169,357,13,29,16,133,133,25,142,188,188,28,191,28,345,92,93,15,57,175,58,23,189,25,59,202,334,180,154],[204,138,68,360,57,58,191,150,331,53,95,56,195,56,172,77,77,146,146,12,98,130,78,138,55,52,169,26,178,336,14,24,11,149,149,28,140,197,197,25,194,25,348,94,93,12,57,172,54,30,196,28,55,183,359,205,193],["PTEN","TRAF3IP2","PCDHB16","PTEN","EP300","SLC17A6","PTEN","FGFR3","RGS7","SLC17A6","CUL3","FLG","CASP8","CDKN2A","ATM","BAP1","PCDHB16","KRAS","TRAF3IP2","ATG16L1","CUL3","HIST1H2AG","CTNNB1","FMN2","PIK3CA","TGFBR2","BAP1","FRG1","VHL","MAP3K1","DNAH5","FRG1","BAP1","FMN2","GPS2","FRG1","FOXQ1","OLFM3","SMAD4","FRG1","RHOA","FRG1","FOXA1","SMARCA4","SMARCA4","PBRM1","SLC17A6","PTEN","KRAS","FRG1","THSD7A","TP53","NFE2L2","EPHA2","CBFB","FBXW7","TP53"],["sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig","sig"],[4.829515113227103,4.867006349392071,4.970726491725888,-2.190072542640713,-2.056995145471575,2.689531002270682,-2.652099290426873,3.79831152160164,-16.60453164736668,-1.266661021034963,0.818117062432978,5.542621053609857,2.908294551809099,6.541829396772697,4.309199585511686,-3.622261734112326,-5.609374077913571,1.896190514457782,-1.623495253265937,-2.895260489975723,2.017285377596438,-2.112184942311468,2.781616199886729,-11.33402020057637,-11.20484438883025,2.955164846727516,-1.471103522258421,-3.285038958518768,3.21880426464522,1.403606141934356,3.867903385418731,3.01218872305513,-4.969941016508062,4.157212347198509,3.228612632856913,7.201614663415379,-5.072718348898515,-2.834582713168035,1.446710975536528,1.957167649336444,1.628501793647038,-1.579917251192453,-4.894743487128653,-8.149878368152576,-3.971345571381814,-7.203124985616852,-5.23030861950003,-8.84585037982284,3.62964356424963,2.609818121400568,6.628021178112419,-2.54281140336302,-7.528379351202729,2.784645555818682,0.470499471736087,0.4696943039986423,-0.3735685268593917],["HNSC","BLCA","LIHC","BRCA","CESC","ESCA","HNSC","BLCA","BRCA","ESCA","KIRP","CESC","HNSC","ESCA","KIRC","LIHC","LIHC","BLCA","BLCA","CHOL","KIRP","BLCA","LIHC","BLCA","CESC","ESCA","KIRC","KICH","KIRC","BRCA","CHOL","KICH","CHOL","BLCA","BLCA","KICH","BLCA","HNSC","HNSC","KICH","HNSC","KICH","BRCA","KIRP","KIRP","CHOL","ESCA","KIRC","CESC","KICH","HNSC","KICH","ESCA","HNSC","BRCA","HNSC","KIRC"],["Allergic_disease","BIP_Cell_2018","BMD_2018","Breast_Cancer","Breast_Cancer","Breast_Cancer","Breast_Cancer","CigarettesPerDay","CigarettesPerDay","IBD_NG2017.hap3","IBD_NG2017.hap3","IQ_NG_2018","IQ_NG_2018","MDD_NG_2018","MDD_NG_2018","Ovarian_cancer","Ovarian_cancer","PD_GWAS_2017_N300k","PD_GWAS_2017_N300k","PD_GWAS_2017_N300k","PD_GWAS_2017_N300k","RA_NG2010.hap3","SCZ_Cell_2018","SmokingCessation","SWB_NG_2015","T2D_DIAGRAM_2017","UC_NG_2015","ukbEUR_ALLERGIC_RHINITIS","ukbEUR_ASTHMA","ukbEUR_BMI","ukbEUR_BMI","ukbEUR_BMI","ukbEUR_CANCER","ukbEUR_CARD","ukbEUR_CARD","ukbEUR_CI_cojo","ukbEUR_DYSLIPID","ukbEUR_DYSLIPID","ukbEUR_DYSLIPID","ukbEUR_DYSLIPID","ukbEUR_Height","ukbEUR_Height","ukbEUR_HYPER","ukbEUR_IRON_DEFICIENCY","ukbEUR_IRRITABLE_BOWEL","ukbEUR_OSTIOA","ukbEUR_OSTIOA","ukbEUR_OSTIOA","ukbEUR_OSTIOP","ukbEUR_OSTIOP","ukbEUR_PSYCHIATRIC","ukbEUR_PVD","ukbEUR_TI_cojo","ukbEUR_VARICOSE_VEINS","Vitiligo","Vitiligo","Vitiligo"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>dd.p<\/th>\n      <th>mlr.p<\/th>\n      <th>mlr.v2.p<\/th>\n      <th>fisher.p<\/th>\n      <th>binom.p<\/th>\n      <th>lr.p<\/th>\n      <th>dd.fdr<\/th>\n      <th>mlr.fdr<\/th>\n      <th>mlr.v2.fdr<\/th>\n      <th>fisher.fdr<\/th>\n      <th>binom.fdr<\/th>\n      <th>lr.fdr<\/th>\n      <th>mut.E1<\/th>\n      <th>mut.E0<\/th>\n      <th>E1<\/th>\n      <th>E0<\/th>\n      <th>gene<\/th>\n      <th>mode<\/th>\n      <th>alpha<\/th>\n      <th>tumor<\/th>\n      <th>pheno<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,19]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"dd.p","targets":1},{"name":"mlr.p","targets":2},{"name":"mlr.v2.p","targets":3},{"name":"fisher.p","targets":4},{"name":"binom.p","targets":5},{"name":"lr.p","targets":6},{"name":"dd.fdr","targets":7},{"name":"mlr.fdr","targets":8},{"name":"mlr.v2.fdr","targets":9},{"name":"fisher.fdr","targets":10},{"name":"binom.fdr","targets":11},{"name":"lr.fdr","targets":12},{"name":"mut.E1","targets":13},{"name":"mut.E0","targets":14},{"name":"E1","targets":15},{"name":"E0","targets":16},{"name":"gene","targets":17},{"name":"mode","targets":18},{"name":"alpha","targets":19},{"name":"tumor","targets":20},{"name":"pheno","targets":21}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>diff_table_reg &lt;- subset(diff_table, mode == &quot;reg&quot;)
datatable(diff_table_reg, options = list(pageLength = 10))</code></pre>
<div class="datatables html-widget html-fill-item" id="htmlwidget-1662b273260d1abeaa23" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-1662b273260d1abeaa23">{"x":{"filter":"none","vertical":false,"data":[["PTEN","TRAF3IP2","PCDHB16","PTEN3","EP300","SLC17A6","RGS7","SLC17A62","CUL3","FLG","ATM","BAP1","PCDHB162","ATG16L1","CUL32","HIST1H2AG","CTNNB1","FMN2","PIK3CA","TGFBR2","BAP12","FRG1","VHL","MAP3K1","DNAH5","BAP13","FMN22","GPS2","FRG13","OLFM3","SMAD4","FRG14","PCDHB7","RHOA","FRG15","FOXA1","SMARCA4","SMARCA42","PBRM1","SLC17A63","PTEN4","KRAS2","THSD7A","TP53","NFE2L2","CBFB","TP532"],[0.00185499089389675,0.000550816198229677,0.00508519812396391,0.00203713553235104,0.0038750600536418,0.00375064492308179,0.00264117269753964,0.0013991870042594,0.0134668851190036,0.00169839976927399,0.00470284526710028,0.0104857566641561,0.0141530970384326,0.0117047169242815,0.00541467719178949,0.00049671600727397,0.00580805177246217,0.000147820127953242,0.0011400182029076,0.00474385412619733,0.00667813091664162,0.0169656248662373,0.009367145992981491,0.00292923235612292,0.0176680280541579,0.0227545596238994,0.00111723743597935,0.00224342028727864,0.0269588698003185,0.00441940763265975,0.00556639703563359,0.00452400484315368,0.00306176062241921,0.000324102052575217,0.0283653200471064,0.00337042810581807,0.00669068669550371,0.0179333225745106,0.0214115427684508,0.00528512866121213,0.00235514848459549,0.00483936445038837,0.00167776743348864,0.0282274271719876,0.00400851991222734,0.00300455005743088,0.008043471786717519],[0.009480431812694709,0.00187916910704073,0.00562380639133483,0.00119571866202735,0.013784216959513,0.0182994478378856,0.0255785449517991,0.000702404424075603,0.0156994960254714,0.000915476819707747,0.00654724841326617,0.009849773890491889,0.0133352194290467,0.0033165280562906,0.0101698348065919,9.86985289390339e-05,0.643959091848326,0.00061665989316527,0.185178972084034,0.010294065373223,0.0247155264895325,0.0291803076198577,0.478459388445093,0.00507396670534522,0.0330448940074625,0.033527032026117,0.00115270326308181,0.00369155275461751,0.0260386254555644,0.00647392784615473,0.00543092302833247,0.00139553546574349,0.053342929297197,0.00146175125851172,0.08497380170214359,0.242182910097343,0.0153472660974998,0.0141933194216884,0.07856491427972589,0.0260599022075523,0.0191766041062386,0.00907752964947005,0.00310699026924578,0.0756249410857557,0.00584905346404875,0.00221828037485174,0.008172077976220561],[0.00796323152095626,0.00446302846042165,0.00539084485966966,0.00321064559110397,0.0107725603662608,0.0320960194386829,0.0247588558873502,0.00103479603995181,0.0180290471432784,0.00277772065286314,0.00651279511632011,0.009531290006496719,0.013622919893334,0.00296969843852933,0.0133472728663123,0.000211669838880817,0.642446217302164,0.00048771999140632,0.0929634947363069,0.0114353355428641,0.0256331357320561,0.012191628219165,0.605401603251718,0.00534431752128926,0.0364362487411359,0.0371384284515621,0.000781475883491265,0.00268668834215391,0.0194804116901591,0.00976910648697529,0.00326580440293922,0.00164169695116618,0.0522739372938578,0.00145042348347773,0.119223219681288,0.25814044512592,0.0161277546608259,0.0149186569856188,0.0803095036613307,0.0379346260508301,0.0195767286835074,0.009528161438274511,0.000861751977136213,0.0126937339892253,0.0067570080584997,0.00208616289542316,0.008287492342394349],[0.0161670151652215,0.00338184782455637,0.500453473607836,0.071172916755035,0.136468545492891,0.057662178300264,0.0122694156197168,0.02508861375739,0.0648062411044901,0.0017891793040132,0.10555784239288,0.0315228997920847,0.498458189733357,0.224137931034483,0.0322550508685066,0.0286207739506133,0.561465980126695,0.00115507746902379,0.217372888696339,0.0671090177742634,0.0201765571657819,0.108695494114916,0.667444399055998,0.06447503219752609,0.227586206896552,0.0433815350389321,0.00269456330148393,0.0312574739417375,0.10426567188387,0.0038387008677736,0.01065584251415,0.126441908394598,0.0607584024322597,0.284020873546915,0.10426567188387,0.0215225626446217,0.119899896033634,0.119050145250955,0.101223581757508,0.3650915385711,0.0147494984001371,0.00425566598867246,0.09710675231521119,0.177226033024544,0.118617614269788,0.00957764517571345,0.07051373888766441],[0.0157947364919782,0.00350438975214083,0.502799759359526,0.07197402936599689,0.14676669534309,0.0286032826921629,0.0118732210941953,0.0217197634640705,0.0353295128905704,0.001953125,0.109506388122296,0.0323153357955876,0.501133786848073,0.197530864197531,0.0308155219853063,0.0282426634731254,0.595910681859219,0.000744971491332067,0.144234878763906,0.0673276745616072,0.0110524880748716,0.06544353112516441,0.577513119554092,0.0658511980937679,0.111619163745364,0.0275496245114698,0.00177018055972546,0.0304627813715476,0.0574166740653867,0.00398324388780982,0.0100788253363646,0.129205218789413,0.0602873230772268,0.176371589288749,0.0574166740653867,0.0224766405393016,0.125151751646913,0.125,0.0390184423106234,0.375,0.00735178511884685,0.00408642005623306,0.07139287272371631,0.154752048576191,0.0541790458811059,0.009337296468258651,0.07044483071770991],[0.0356093764068048,0.027646990920746,0.988028344316621,0.0786834537637577,0.0468548326211326,0.988910804594108,0.978954444184203,0.988837755315,0.0825508323469696,0.989891971836624,0.118073669175601,0.986519264221453,0.987985470954869,0.994874206248387,0.07221163036316219,0.06718693204165641,0.419475607533723,0.0034203650883374,0.101883110414749,0.987751438460023,0.0195668612611154,0.10793579022121,0.738660335798048,0.120954958614628,0.993746331929972,0.993621741820586,0.00597721744142133,0.111355381735404,0.09233781705187299,0.98347578387919,0.0422292505386247,0.145042151726466,0.120368650376618,0.167566614601912,0.09233781705187311,0.0257681484000709,0.13996696143815,0.134649520185605,0.995304032941256,0.332154861499259,0.977339654199706,0.989313329335114,0.0333482859776124,0.111807399549558,0.990386553211336,0.0299563341915053,0.98315839814772],[0.0612146994985927,0.0347014204884697,0.0661075756115309,0.0509283883087759,0.06587602091191069,0.0375064492308179,0.06602931743849109,0.013991870042594,0.0538675404760143,0.0288727960776578,0.0470284526710028,0.09199513074981169,0.09199513074981169,0.0468188676971262,0.021658708767158,0.0312931084582601,0.0755046730420082,0.00931266806105422,0.0193803094494292,0.0474385412619733,0.0667813091664162,0.0339312497324746,0.0936714599298149,0.073230808903073,0.0706721122166315,0.09101823849559761,0.0703859584666993,0.0706677390492771,0.053917739600637,0.09184555108795429,0.09184555108795429,0.00904800968630735,0.07654401556048029,0.0106953677349821,0.0567306400942127,0.0842607026454518,0.0267627467820148,0.0717332902980424,0.0856461710738033,0.0528512866121213,0.0235514848459549,0.0822691956566023,0.055366325305125,0.0564548543439752,0.0400851991222734,0.07511375143577211,0.0804347178671752],[0.312854249818925,0.118387653743566,0.0731094830873527,0.0298929665506837,0.234331688311721,0.182994478378856,0.211411901641226,0.00702404424075603,0.0627979841018856,0.00778155296751585,0.0327362420663309,0.08667892628880321,0.08667892628880321,0.0132661122251624,0.0406793392263676,0.00621800732315914,0.901749954393567,0.038849573269412,0.44972036077551,0.10294065373223,0.247155264895325,0.0583606152397155,0.830729660147508,0.12684916763363,0.13217957602985,0.134108128104468,0.072620305574154,0.116283911770451,0.0520772509111288,0.0712132063077021,0.0712132063077021,0.00279107093148698,0.82093505849121,0.0482377915308866,0.169947603404287,0.713039957397685,0.061389064389999,0.0567732776867534,0.314259657118904,0.260599022075523,0.10386984444119,0.154318004040991,0.102530678885111,0.151249882171511,0.0584905346404875,0.0554570093712934,0.0817207797622056],[0.262786640191556,0.281170793006564,0.0700809831757056,0.08026613977759919,0.183133526226434,0.320960194386829,0.199392380986264,0.0103479603995181,0.0721161885731134,0.0236106255493367,0.0325639755816005,0.0885489793066709,0.0885489793066709,0.0118787937541173,0.0533890914652491,0.0133351998494915,0.901477215807779,0.0307263594585982,0.316075882103443,0.114353355428641,0.256331357320561,0.02438325643833,0.839315605943044,0.133607938032231,0.145744994964544,0.148553713806248,0.0492329806599497,0.0846306827778481,0.0389608233803181,0.107460171356728,0.107460171356728,0.00328339390233237,0.7720580070352659,0.0478639749547652,0.238446439362575,0.680995438961347,0.06451101864330371,0.0596746279424752,0.321238014645323,0.379346260508301,0.154914031066674,0.161978744450667,0.028437815245495,0.0253874679784506,0.067570080584997,0.0521540723855791,0.08287492342394349],[0.505168555110707,0.07101880431568371,1,0.444830729718969,0.773321757793049,0.57662178300264,0.306735390492919,0.155181561831178,0.259224964417961,0.0304160481682243,0.6630601021439519,0.359370638695053,0.877943446802507,0.620721436516765,0.129020203474026,0.42370333471841,0.856917961555913,0.07276988054849889,0.739067821567552,0.620317573401351,0.201765571657819,0.217390988229833,1,0.439827873914768,0.9103448275862061,0.173526140155729,0.169757487993487,0.218802317592162,0.20853134376774,0.126677128636529,0.175821401483475,0.180760032063965,0.759480030403247,0.935676699201717,0.20853134376774,0.538064066115543,0.479599584134538,0.476200581003821,0.404894327030033,0.91272884642775,0.147494984001371,0.07234632180743179,1,0.354452066049087,0.871047531068365,0.239441129392836,0.705137388876644],[0.496935483387718,0.07359218479495749,1,0.449837683537481,0.818205299960959,0.286032826921629,0.296830527354882,0.144995789303663,0.141318051562282,0.033203125,0.538602404388952,0.334222880384202,0.878270885073606,0.6342021033379061,0.123262087941225,0.311718879632863,0.863482879650901,0.0469332039539202,0.490398587797279,0.616826086282649,0.110524880748716,0.130887062250329,1,0.445733265194503,0.446476654981456,0.110198498045879,0.111521375262704,0.19191552264075,0.114833348130773,0.131447048297724,0.166300618050017,0.129205218789413,0.753591538465335,0.889433273626956,0.114833348130773,0.382242801416714,0.500607006587652,0.5,0.156073769242493,0.890409338093152,0.0735178511884685,0.06946914095596191,1,0.309504097152382,0.541790458811059,0.233432411706466,0.7044483071770991],[0.636178931435901,0.43544010700175,0.988028344316621,0.491771586023486,0.51224732519535,0.988910804594108,0.978954444184203,0.988837755315,0.330203329387878,1,0.594236493120462,0.988880913437802,0.988880913437802,0.994874206248387,0.288846521452649,0.705462786437392,0.81404824697231,0.215483000565256,0.988958787226045,0.990532016217221,0.195668612611154,0.21587158044242,0.83900700932446,0.755968491341423,0.993746331929972,0.993922218955287,0.376564698809544,0.690383244416802,0.184675634103746,0.98347578387919,0.574340626340533,0.145042151726466,0.724766084307879,0.9217993423538851,0.184675634103746,0.644203710001772,0.559867845752599,0.538598080742422,0.995304032941256,0.664309722998518,0.977339654199706,0.993700990295169,0.821078609788877,0.223614799099116,0.990386553211336,0.643492672691279,0.98315839814772],[8,12,2,1,3,5,0,0,7,10,8,0,0,0,7,1,13,4,11,5,2,1,59,14,3,0,19,7,6,0,9,6,6,6,1,2,1,1,0,1,0,9,17,5,0,9,0],[1,1,0,7,9,0,6,5,1,0,2,6,2,2,1,7,19,20,19,0,11,6,56,5,0,4,5,1,1,9,1,1,1,2,6,11,6,6,4,4,7,0,8,13,4,1,5],[181,144,79,333,55,56,362,61,91,56,175,70,70,15,88,152,69,144,57,62,178,27,169,357,13,16,133,133,25,188,188,28,332,191,28,345,92,93,15,57,175,58,189,25,59,334,154],[204,138,68,360,57,58,331,53,95,56,172,77,77,12,98,130,78,138,55,52,169,26,178,336,14,11,149,149,28,197,197,25,361,194,25,348,94,93,12,57,172,54,196,28,55,359,193],["PTEN","TRAF3IP2","PCDHB16","PTEN","EP300","SLC17A6","RGS7","SLC17A6","CUL3","FLG","ATM","BAP1","PCDHB16","ATG16L1","CUL3","HIST1H2AG","CTNNB1","FMN2","PIK3CA","TGFBR2","BAP1","FRG1","VHL","MAP3K1","DNAH5","BAP1","FMN2","GPS2","FRG1","OLFM3","SMAD4","FRG1","PCDHB7","RHOA","FRG1","FOXA1","SMARCA4","SMARCA4","PBRM1","SLC17A6","PTEN","KRAS","THSD7A","TP53","NFE2L2","CBFB","TP53"],["reg","reg","reg","reg","reg","reg","reg","reg","reg","reg","reg","reg","reg","reg","reg","reg","reg","reg","reg","reg","reg","reg","reg","reg","reg","reg","reg","reg","reg","reg","reg","reg","reg","reg","reg","reg","reg","reg","reg","reg","reg","reg","reg","reg","reg","reg","reg"],[4.885387913972101,4.70782386332527,4.942880743793217,-2.204017879682629,-2.055225511834662,2.601949097127149,-16.801248269044,-1.276059169476036,0.8201743301856167,5.732453921397108,4.344752017591009,-3.672861259347021,-5.373896131096783,-2.828402237488384,2.043552759920137,-2.058382718261794,2.753652055660934,-10.80970565992324,-12.25884478176203,2.990832011939489,-1.669647798875427,-3.187109920906896,3.208868238019873,1.356134839068513,3.791332341120615,-4.988660731466308,4.030418545084753,3.206078667581179,7.080834866482602,-2.808826176555622,1.465079365990379,1.941288287634664,2.050448387669908,1.621764437118898,-1.635161687328421,-4.921390424472267,-8.089484456378749,-3.970941630235973,-6.777205589232756,-5.348556309374477,-8.813060987276003,3.370612261369379,6.629229647761257,-2.589185647046574,-7.760398765440986,0.5239667074690195,-0.3719848466791491],["HNSC","BLCA","LIHC","BRCA","CESC","ESCA","BRCA","ESCA","KIRP","CESC","KIRC","LIHC","LIHC","CHOL","KIRP","BLCA","LIHC","BLCA","CESC","ESCA","KIRC","KICH","KIRC","BRCA","CHOL","CHOL","BLCA","BLCA","KICH","HNSC","HNSC","KICH","BRCA","HNSC","KICH","BRCA","KIRP","KIRP","CHOL","ESCA","KIRC","CESC","HNSC","KICH","ESCA","BRCA","KIRC"],["Allergic_disease","BIP_Cell_2018","BMD_2018","Breast_Cancer","Breast_Cancer","Breast_Cancer","CigarettesPerDay","IBD_NG2017.hap3","IBD_NG2017.hap3","IQ_NG_2018","MDD_NG_2018","Ovarian_cancer","Ovarian_cancer","PD_GWAS_2017_N300k","PD_GWAS_2017_N300k","RA_NG2010.hap3","SCZ_Cell_2018","SmokingCessation","SWB_NG_2015","T2D_DIAGRAM_2017","UC_NG_2015","ukbEUR_ALLERGIC_RHINITIS","ukbEUR_ASTHMA","ukbEUR_BMI","ukbEUR_BMI","ukbEUR_CANCER","ukbEUR_CARD","ukbEUR_CARD","ukbEUR_CI_cojo","ukbEUR_DYSLIPID","ukbEUR_DYSLIPID","ukbEUR_DYSLIPID","ukbEUR_Height","ukbEUR_Height","ukbEUR_Height","ukbEUR_HYPER","ukbEUR_IRON_DEFICIENCY","ukbEUR_IRRITABLE_BOWEL","ukbEUR_OSTIOA","ukbEUR_OSTIOA","ukbEUR_OSTIOA","ukbEUR_OSTIOP","ukbEUR_PSYCHIATRIC","ukbEUR_PVD","ukbEUR_TI_cojo","Vitiligo","Vitiligo"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>dd.p<\/th>\n      <th>mlr.p<\/th>\n      <th>mlr.v2.p<\/th>\n      <th>fisher.p<\/th>\n      <th>binom.p<\/th>\n      <th>lr.p<\/th>\n      <th>dd.fdr<\/th>\n      <th>mlr.fdr<\/th>\n      <th>mlr.v2.fdr<\/th>\n      <th>fisher.fdr<\/th>\n      <th>binom.fdr<\/th>\n      <th>lr.fdr<\/th>\n      <th>mut.E1<\/th>\n      <th>mut.E0<\/th>\n      <th>E1<\/th>\n      <th>E0<\/th>\n      <th>gene<\/th>\n      <th>mode<\/th>\n      <th>alpha<\/th>\n      <th>tumor<\/th>\n      <th>pheno<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,19]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"dd.p","targets":1},{"name":"mlr.p","targets":2},{"name":"mlr.v2.p","targets":3},{"name":"fisher.p","targets":4},{"name":"binom.p","targets":5},{"name":"lr.p","targets":6},{"name":"dd.fdr","targets":7},{"name":"mlr.fdr","targets":8},{"name":"mlr.v2.fdr","targets":9},{"name":"fisher.fdr","targets":10},{"name":"binom.fdr","targets":11},{"name":"lr.fdr","targets":12},{"name":"mut.E1","targets":13},{"name":"mut.E0","targets":14},{"name":"E1","targets":15},{"name":"E0","targets":16},{"name":"gene","targets":17},{"name":"mode","targets":18},{"name":"alpha","targets":19},{"name":"tumor","targets":20},{"name":"pheno","targets":21}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>plot_diff_number &lt;- function(filedf, mode = c(&quot;all&quot;, &quot;reg&quot;, &quot;sig&quot;)){
  mode &lt;- match.arg(mode)
  
  if (mode != &quot;all&quot;){
    filedf &lt;- filedf[filedf$mode == mode, ]
  }
  
  pheno_all &lt;- unique(filedf$phenotype)
  numlist &lt;- list()
  
  for (p in pheno_all){
    p_txtfiles &lt;- filedf[filedf$phenotype == p, ]
    numlist[[p]] &lt;- list()
    for (t in 1:nrow(p_txtfiles)){
      txtf &lt;- p_txtfiles[t, &quot;filetxt&quot;]
      tumor &lt;- p_txtfiles[t, &quot;tumor&quot;]
      res &lt;- read.table(txtf, header = TRUE)
      numlist[[p]][[tumor]] &lt;- c(
        nrow(res[res$dd.fdr &lt; 0.1,]),
        nrow(res) - nrow(res[res$dd.fdr &lt; 0.1,])
      )
    }
  }
  
  par(mfrow = c(length(numlist),1), mar = c(0,0,0,0))
  for (phenotype in names(numlist)) {
    colors &lt;- rainbow(length(numlist[[phenotype]]))
    plot(
      NULL,
      xlim = c(0.5, length(numlist[[phenotype]]) + 0.5),
      ylim = c(0, max(unlist(sapply(numlist[[phenotype]], `[`, 1)))),
      xlab = &quot;Tumor Type&quot;,
      ylab = &quot;No. Genes&quot;,
      main = paste(&quot;Context:&quot;, phenotype),
      xaxt = &quot;n&quot;, bty = &#39;n&#39;
    )
    grid()
    
    axis(
      1,
      at = 1:length(numlist[[phenotype]]),
      labels = names(numlist[[phenotype]]),
      las = 2
    )
    
    for (i in seq_along(numlist[[phenotype]])) {
      bar_height &lt;- numlist[[phenotype]][[i]][1]
      rect(i - 0.4, 0, i + 0.4, bar_height, col = colors[i], border = NA)
    }
  }
}</code></pre>
<pre class="r"><code>plot_diff_number(filedf, mode = &quot;reg&quot;)</code></pre>
<p><img src="figure/PRS.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LD-ZhangQR/diffdriver-analysis/blob/65df6097c11c1e62f4317b077bffa94e99c1f7c6/docs/figure/PRS.Rmd/unnamed-chunk-7-1.png" target="_blank">65df609</a>
</td>
<td>
Qirui Zhang
</td>
<td>
2025-04-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_diff_number(filedf, mode = &quot;sig&quot;)</code></pre>
<p><img src="figure/PRS.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LD-ZhangQR/diffdriver-analysis/blob/65df6097c11c1e62f4317b077bffa94e99c1f7c6/docs/figure/PRS.Rmd/unnamed-chunk-8-1.png" target="_blank">65df609</a>
</td>
<td>
Qirui Zhang
</td>
<td>
2025-04-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_each &lt;- function(outdt, path.base){
  options(repr.plot.width = 7, repr.plot.height = 3)
  
  for (i in seq_len(nrow(outdt))) {

    hits_i &lt;- paste0(
      &quot;testdiffdriver_&quot;,
      outdt$mode[i], &quot;_&quot;,
      outdt$pheno[i],
      &quot;_resdd.Rd&quot;
    )
    
    subfolder &lt;- paste0(outdt$pheno[i], &quot;_&quot;, outdt$tumor[i], &quot;_mutations&quot;)
    path &lt;- file.path(path.base, outdt$pheno[i], subfolder, hits_i)
    
    load(path)
    ganno &lt;- fannoallg[[ outdt$gene[i] ]]
    cannoj &lt;- e
    names(cannoj) &lt;- &quot;Phenotype&quot;
    mutmtx &lt;- riallg[[ outdt$gene[i] ]] 
    bmrmtx &lt;- bmrallg[[ outdt$gene[i] ]]
    
    message(&quot;Plotting: &quot;, outdt$tumor[i], &quot;_&quot;, outdt$gene[i], &quot;_&quot;, outdt$pheno[i], &quot;_&quot;, outdt$mode[i])
    
    colnames(mutmtx)[colnames(mutmtx) == &quot;chrom&quot;] &lt;- &quot;Chromosome&quot;
    colnames(mutmtx)[colnames(mutmtx) == &quot;start&quot;] &lt;- &quot;Start&quot;
    colnames(mutmtx)[colnames(mutmtx) == &quot;ref&quot;]   &lt;- &quot;Ref&quot;
    colnames(mutmtx)[colnames(mutmtx) == &quot;alt&quot;]   &lt;- &quot;Alt&quot;

    diffdriver::plot_mut(
      gene_name   = outdt$gene[i], 
      mut         = mutmtx,        
      pheno       = cannoj,       
      totalnttype = 96,         
      anno_dir    = &quot;/dartfs/rc/lab/S/Szhao/qiruiz/diffdriver/temp/annodir96&quot;,
      output_prefix = paste(
      outdt$tumor[i], outdt$gene[i], outdt$pheno[i], outdt$mode[i],
      sep = &quot;_&quot;
      ),
      output_dir = &quot;/dartfs/rc/lab/S/Szhao/qiruiz/diffdriver/workflowr/diffdriver/output/&quot;
    )
    
    diffdriver::plot_mut(gene_name = &quot;PIK3CA&quot;, mut= mut, pheno = pheno, totalnttype = 9, anno_dir = &quot;~/temp/annodir9&quot;)
  
  }
}</code></pre>
<pre class="r"><code>outdt &lt;- data.frame(
  tumor = c(&quot;BLCA&quot;, &quot;BLCA&quot;),
  pheno = c(&quot;Aneuploidy.Score&quot;, &quot;Aneuploidy.Score&quot;),
  gene  = c(&quot;FGFR3&quot;, &quot;NFE2L2&quot;),
  mode  = c(&quot;reg&quot;, &quot;reg&quot;)
)

##plot_each(
##  outdt     = outdt,
##  path.base = &quot;/dartfs/rc/lab/S/Szhao/qiruiz/diffdriver/temp/output&quot;
##)</code></pre>
<pre class="r"><code>mut_path &lt;- &quot;/dartfs/rc/lab/S/Szhao/qiruiz/diffdriver/tumor_specific_input/HNSC/HNSC_mutations.txt&quot;
mut &lt;- read.table(mut_path, header = TRUE, sep = &quot;\t&quot;)
head(mut)</code></pre>
<pre><code>  Chromosome Position Ref Alt     SampleID
1          4 44700691   C   G TCGA-CR-7386
2          9 35074164   C   T TCGA-CR-7395
3         22 37914082   T   A TCGA-CQ-5325
4          6 30571952   C   T TCGA-CR-6467
5         19 38127001   C   G TCGA-CV-5434
6          8 23290623   C   T TCGA-HD-7754</code></pre>
<pre class="r"><code>pheno_path &lt;- &quot;/dartfs/rc/lab/S/Szhao/qiruiz/diffdriver/batch5/tumor_specific_input/HNSC/HNSC_PRS_46phenotype.txt&quot;
pheno &lt;- read.table(pheno_path, header = TRUE, sep = &quot;\t&quot;)
head(pheno)</code></pre>
<pre><code>      SampleID Nsyn BMRlabel IQ_NG_2018 RA_NG2010.hap3 SmokingCessation
1 TCGA-BA-4074   31     HNSC    1.45831       0.694409       -0.3981500
2 TCGA-BA-4076   75     HNSC    1.29510       1.823400        0.2036340
3 TCGA-BA-4078  108     HNSC    1.07782       2.100250        0.4714600
4 TCGA-BA-5149   20     HNSC    1.89861       0.479005        1.0458500
5 TCGA-BA-5152  131     HNSC    2.29903       1.595350        0.2975740
6 TCGA-BA-5153    5     HNSC    1.72524       1.430420       -0.0655172
  EA_NG_2018_excluding_23andMe    Vitiligo SmokingInitiation MDD_NG_2018
1                    -1.238970  1.26864000         -1.050170    0.382579
2                    -1.154450  2.45535000         -0.961242    0.410053
3                    -1.342810 -0.00148262         -0.844708    0.733341
4                    -0.988011  0.92995000         -0.851027    0.548987
5                    -1.110470  5.76225000         -0.806739    0.533504
6                    -1.741480  2.22627000         -0.710927    1.352270
  DrinksPerWeek UC_NG_2015 SWB_NG_2015  BMD_2018 Stroke_GWAS_2018
1     -0.521843   -7.14847  -0.0150747 -0.835407         -5.63974
2     -0.358751   -7.55253   0.0682952 -0.249712         -5.90477
3     -0.449110   -6.68307  -0.0256169 -0.581139         -6.00953
4     -0.329349   -8.60876   0.1674810 -0.486727         -5.35685
5     -0.465745   -7.33514  -0.1184330 -0.729187         -5.61845
6     -0.309259   -7.10804  -0.2327200 -0.878936         -5.73646
  T2D_DIAGRAM_2017 BIP_Cell_2018 SCZ_Cell_2018 PD_GWAS_2017_N300k
1         -3.32332      1.014470      1.040450           -2.54724
2         -2.51051      1.224600      1.387370           -2.31817
3         -4.66238      1.888960      1.005550           -1.88996
4         -2.67323      0.165109      1.443750           -1.85760
5         -1.59984      0.627479      0.944975           -2.64566
6         -1.83947      1.767970      2.351110           -1.78856
  IBD_NG2017.hap3 Breast_Cancer CigarettesPerDay Allergic_disease
1      -0.1333830     -1.899220        0.2879450        -0.530691
2       0.0339394     -0.450438       -0.0837119        -0.526251
3       0.7195770     -0.420934       -0.2693660        -0.782519
4       1.4229300     -1.672410        0.2281060        -1.222970
5       0.0034194     -0.217626        0.0877931        -1.602410
6       1.5727800     -0.195844        0.1722290        -1.490610
  Ovarian_cancer Prostate_cancer_2017_common ukbEUR_ASTHMA ukbEUR_Height
1     -0.8154990                   -2.066160    -0.5625660       4.25508
2     -0.7184930                   -1.542730     0.0825560       2.23646
3     -0.8621700                   -0.914401    -1.4001800       4.93102
4     -0.1092760                   -1.423450     0.9943640       3.53095
5      0.0492364                   -1.619290    -0.0544336       3.60281
6      0.0916813                   -1.308620    -0.4393400       4.40934
  ukbEUR_CASES ukbEUR_OBESITY ukbEUR_IRON_DEFICIENCY ukbEUR_CANCER
1   -0.5170490      -3.732770              0.1646340     0.1302910
2    0.5811700      -4.014110              0.1304370     0.3097460
3   -1.1311400      -3.933410             -0.1622100    -0.0580030
4   -0.0186488      -2.426730              0.0473272     0.1627710
5   -0.0606553      -0.736967              0.0289600     0.0953033
6   -0.1117660      -1.423700              0.2528540     0.1641480
  ukbEUR_VARICOSE_VEINS ukbEUR_BMI ukbEUR_OSTIOA ukbEUR_HYPER ukbEUR_CI_cojo
1               2.35291   -2.30652      0.119207    -0.253161      0.0754179
2               3.12558   -2.92003      1.012030    -0.397370     -0.1155150
3               3.16937   -2.76420     -0.220231    -1.210220      0.1514880
4               3.35101   -1.74280      0.361837     0.110489     -0.1751540
5               3.50664   -1.25031      0.152059     0.693521     -0.1704860
6               2.15890   -2.21941      0.798282     2.329600     -0.1995070
  ukbEUR_DIA2 ukbEUR_IRRITABLE_BOWEL ukbEUR_HERNIA_ABDOMINOPELVIC
1    0.765633               0.541714                     0.503483
2   -1.751520               0.693633                     1.087900
3   -0.543828               1.151110                     0.730177
4    1.215080               0.493616                     1.376230
5    0.962090               0.281611                     0.340649
6    3.311180               0.551926                     0.994038
  ukbEUR_SEVERE_OBESITY ukbEUR_CARD ukbEUR_DYSLIPID ukbEUR_TI_cojo
1               4.63876   0.0300777        1.493630      0.0590074
2               2.41792   0.3174030        0.229772     -0.3132760
3               3.77742  -0.3389840       -0.180234     -0.3755540
4               7.06223  -0.0830913       -0.951996     -0.5700890
5               5.81562   0.3711490        0.427656     -0.5463310
6               4.12682   0.7286250        1.225030     -0.0910027
  ukbEUR_PEPTIC_ULCERS ukbEUR_ALLERGIC_RHINITIS ukbEUR_PSYCHIATRIC ukbEUR_PVD
1             0.168048               -0.1366340         -0.1945100 -0.1945590
2             0.700666                1.4910600         -0.2802240  0.4111770
3             0.345579                0.8357870         -0.3356700 -0.0968104
4             0.277767               -0.0600145         -0.0681533  0.6169340
5             0.026319               -0.2695450          0.4114150  0.1468530
6             0.574635                0.2389460          0.2802970  0.3452900
  ukbEUR_OSTIOP ukbEUR_HEMORRHOIDS IQ_NG_2018.1
1      0.799799           0.207470      1.45831
2     -0.438758           0.527223      1.29510
3      0.397068           0.756136      1.07782
4      1.128380           0.359372      1.89861
5     -0.186162           0.753357      2.29903
6     -0.421300           0.565560      1.72524</code></pre>
<pre class="r"><code>diffdriver::plot_mut(gene_name = &quot;CDKN2A&quot;, mut= mut, pheno = pheno, totalnttype = 96, 
                     anno_dir =  &quot;/dartfs/rc/lab/S/Szhao/qiruiz/diffdriver/temp/annodir96&quot;)</code></pre>
<pre><code>[1] &quot;coding...&quot;
[1] &quot;processing ...&quot;
[1] &quot;for qnvars, filling in missing values ...&quot;
[1] &quot;for cvars (0/1 categories), filling in missing values ...&quot;
[1] &quot;normalizing categorical variables in annotation matrix ...&quot;
[1] &quot;Only keeping the first two columns of the phenotype data frame.&quot;
[1] &quot;phenotype name is Nsyn&quot;
[1] &quot;number of samples shared in phenotype and mutation file: 385&quot;</code></pre>
<p><img src="figure/PRS.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LD-ZhangQR/diffdriver-analysis/blob/65df6097c11c1e62f4317b077bffa94e99c1f7c6/docs/figure/PRS.Rmd/unnamed-chunk-13-1.png" target="_blank">65df609</a>
</td>
<td>
Qirui Zhang
</td>
<td>
2025-04-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>library(dplyr)</code></pre>
<pre><code>
Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    filter, lag</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    intersect, setdiff, setequal, union</code></pre>
<pre class="r"><code>df_summary &lt;- diff_table %&gt;%
  filter(mode == &quot;sig&quot;) %&gt;%               
  group_by(tumor, pheno) %&gt;%          
  summarise(num_genes = n_distinct(gene)) %&gt;% 
  ungroup()</code></pre>
<pre><code>`summarise()` has grouped output by &#39;tumor&#39;. You can override using the
`.groups` argument.</code></pre>
<pre class="r"><code>head(df_summary)</code></pre>
<pre><code># A tibble: 6 × 3
  tumor pheno              num_genes
  &lt;chr&gt; &lt;chr&gt;                  &lt;int&gt;
1 BLCA  BIP_Cell_2018              1
2 BLCA  CigarettesPerDay           1
3 BLCA  PD_GWAS_2017_N300k         2
4 BLCA  RA_NG2010.hap3             1
5 BLCA  SmokingCessation           1
6 BLCA  ukbEUR_CARD                2</code></pre>
<pre class="r"><code>plot_one_pheno &lt;- function(df_single, pheno_name) {

  par(mfrow=c(1,1), mar = c(3,5,2,0))
  
  colors &lt;- rainbow(nrow(df_single))
  ymax   &lt;- max(df_single$num_genes)
  
  plot(
    NULL,
    xlim = c(0.5, nrow(df_single) + 0.5),
    ylim = c(0, ymax),
    xlab = &quot;Tumor Type&quot;,
    ylab = &quot;No. Genes&quot;,
    main = paste(&quot;Context:&quot;, pheno_name),
    xaxt = &quot;n&quot;, bty = &#39;n&#39;
  )
  grid()
  axis(
    1,
    at = 1:nrow(df_single),
    labels = df_single$tumor,
    las=2
  )
  
  for(i in seq_len(nrow(df_single))) {
    rect(i-0.4, 0, i+0.4, df_single$num_genes[i],
         col=colors[i], border=NA)
  }
}

df_single &lt;- subset(df_summary, pheno == &quot;BIP_Cell_2018&quot;)
plot_one_pheno(
  df_single  = df_single, 
  pheno_name = &quot;BIP_Cell_2018&quot; 
)</code></pre>
<p><img src="figure/PRS.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LD-ZhangQR/diffdriver-analysis/blob/65df6097c11c1e62f4317b077bffa94e99c1f7c6/docs/figure/PRS.Rmd/unnamed-chunk-15-1.png" target="_blank">65df609</a>
</td>
<td>
Qirui Zhang
</td>
<td>
2025-04-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df_table &lt;- diff_table %&gt;%
  filter(mode == &quot;sig&quot;) %&gt;%
  transmute(
    Tumor   = tumor,
    Context = pheno,
    Gene    = gene,
    p       = dd.p,
    FDR     = dd.fdr
  ) %&gt;%
  filter(FDR &lt; 0.1) %&gt;%
  arrange(FDR)

head(df_table)</code></pre>
<pre><code>        Tumor          Context   Gene            p          FDR
MAP3K11  BRCA       ukbEUR_BMI MAP3K1 3.871623e-06 9.679056e-05
FRG112   KICH  ukbEUR_DYSLIPID   FRG1 3.827380e-03 7.654760e-03
FMN21    BLCA SmokingCessation   FMN2 1.220992e-04 7.692249e-03
PTEN2    HNSC    Breast_Cancer   PTEN 2.442169e-04 8.059158e-03
PIK3CA1  CESC      SWB_NG_2015 PIK3CA 5.159051e-04 8.770387e-03
RHOA1    HNSC    ukbEUR_Height   RHOA 2.848441e-04 9.399854e-03</code></pre>
<pre class="r"><code>library(gt)

df_table %&gt;%
  head(8) %&gt;%
  gt() %&gt;%
  tab_header(
    title = &quot;Significant Drivers&quot;,
    subtitle = &quot;FDR &lt; 0.1&quot;
  ) %&gt;%

  tab_style(
    style = list(
      cell_fill(color = &quot;#B60059&quot;), 
      cell_text(color = &quot;white&quot;, weight = &quot;bold&quot;)
    ),
    locations = cells_column_labels(everything())
  ) %&gt;%
  fmt_scientific(
    columns = c(p, FDR),
    decimals = 2
  ) %&gt;%
  tab_options(
    table.border.top.style = &quot;none&quot;,
    table.border.bottom.style = &quot;none&quot;,
    table.border.left.style = &quot;none&quot;,
    table.border.right.style = &quot;none&quot;,
    
    column_labels.border.bottom.style = &quot;solid&quot;,
    column_labels.border.bottom.width = px(2),
    column_labels.border.bottom.color = &quot;black&quot;,
    
    data_row.padding = px(4),   
    table.width = pct(50)           
  )</code></pre>
<div id="smsmddwcst" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#smsmddwcst table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#smsmddwcst thead, #smsmddwcst tbody, #smsmddwcst tfoot, #smsmddwcst tr, #smsmddwcst td, #smsmddwcst th {
  border-style: none;
}

#smsmddwcst p {
  margin: 0;
  padding: 0;
}

#smsmddwcst .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: 50%;
  border-top-style: none;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#smsmddwcst .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#smsmddwcst .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#smsmddwcst .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#smsmddwcst .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#smsmddwcst .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#smsmddwcst .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #000000;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#smsmddwcst .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#smsmddwcst .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#smsmddwcst .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#smsmddwcst .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#smsmddwcst .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #000000;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#smsmddwcst .gt_spanner_row {
  border-bottom-style: hidden;
}

#smsmddwcst .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#smsmddwcst .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#smsmddwcst .gt_from_md > :first-child {
  margin-top: 0;
}

#smsmddwcst .gt_from_md > :last-child {
  margin-bottom: 0;
}

#smsmddwcst .gt_row {
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#smsmddwcst .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#smsmddwcst .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#smsmddwcst .gt_row_group_first td {
  border-top-width: 2px;
}

#smsmddwcst .gt_row_group_first th {
  border-top-width: 2px;
}

#smsmddwcst .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#smsmddwcst .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#smsmddwcst .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#smsmddwcst .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#smsmddwcst .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#smsmddwcst .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#smsmddwcst .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#smsmddwcst .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#smsmddwcst .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#smsmddwcst .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#smsmddwcst .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#smsmddwcst .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#smsmddwcst .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#smsmddwcst .gt_left {
  text-align: left;
}

#smsmddwcst .gt_center {
  text-align: center;
}

#smsmddwcst .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#smsmddwcst .gt_font_normal {
  font-weight: normal;
}

#smsmddwcst .gt_font_bold {
  font-weight: bold;
}

#smsmddwcst .gt_font_italic {
  font-style: italic;
}

#smsmddwcst .gt_super {
  font-size: 65%;
}

#smsmddwcst .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#smsmddwcst .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#smsmddwcst .gt_indent_1 {
  text-indent: 5px;
}

#smsmddwcst .gt_indent_2 {
  text-indent: 10px;
}

#smsmddwcst .gt_indent_3 {
  text-indent: 15px;
}

#smsmddwcst .gt_indent_4 {
  text-indent: 20px;
}

#smsmddwcst .gt_indent_5 {
  text-indent: 25px;
}

#smsmddwcst .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#smsmddwcst div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>
<table class="gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
  <thead>
    <tr class="gt_heading">
      <td colspan="5" class="gt_heading gt_title gt_font_normal" style>Significant Drivers</td>
    </tr>
    <tr class="gt_heading">
      <td colspan="5" class="gt_heading gt_subtitle gt_font_normal gt_bottom_border" style>FDR &lt; 0.1</td>
    </tr>
    <tr class="gt_col_headings">
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" style="background-color: #B60059; color: #FFFFFF; font-weight: bold;" scope="col" id="Tumor">Tumor</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" style="background-color: #B60059; color: #FFFFFF; font-weight: bold;" scope="col" id="Context">Context</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" style="background-color: #B60059; color: #FFFFFF; font-weight: bold;" scope="col" id="Gene">Gene</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" style="background-color: #B60059; color: #FFFFFF; font-weight: bold;" scope="col" id="p">p</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" style="background-color: #B60059; color: #FFFFFF; font-weight: bold;" scope="col" id="FDR">FDR</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><td headers="Tumor" class="gt_row gt_left">BRCA</td>
<td headers="Context" class="gt_row gt_left">ukbEUR_BMI</td>
<td headers="Gene" class="gt_row gt_left">MAP3K1</td>
<td headers="p" class="gt_row gt_right">3.87&nbsp;×&nbsp;10<sup style='font-size: 65%;'>−6</sup></td>
<td headers="FDR" class="gt_row gt_right">9.68&nbsp;×&nbsp;10<sup style='font-size: 65%;'>−5</sup></td></tr>
    <tr><td headers="Tumor" class="gt_row gt_left">KICH</td>
<td headers="Context" class="gt_row gt_left">ukbEUR_DYSLIPID</td>
<td headers="Gene" class="gt_row gt_left">FRG1</td>
<td headers="p" class="gt_row gt_right">3.83&nbsp;×&nbsp;10<sup style='font-size: 65%;'>−3</sup></td>
<td headers="FDR" class="gt_row gt_right">7.65&nbsp;×&nbsp;10<sup style='font-size: 65%;'>−3</sup></td></tr>
    <tr><td headers="Tumor" class="gt_row gt_left">BLCA</td>
<td headers="Context" class="gt_row gt_left">SmokingCessation</td>
<td headers="Gene" class="gt_row gt_left">FMN2</td>
<td headers="p" class="gt_row gt_right">1.22&nbsp;×&nbsp;10<sup style='font-size: 65%;'>−4</sup></td>
<td headers="FDR" class="gt_row gt_right">7.69&nbsp;×&nbsp;10<sup style='font-size: 65%;'>−3</sup></td></tr>
    <tr><td headers="Tumor" class="gt_row gt_left">HNSC</td>
<td headers="Context" class="gt_row gt_left">Breast_Cancer</td>
<td headers="Gene" class="gt_row gt_left">PTEN</td>
<td headers="p" class="gt_row gt_right">2.44&nbsp;×&nbsp;10<sup style='font-size: 65%;'>−4</sup></td>
<td headers="FDR" class="gt_row gt_right">8.06&nbsp;×&nbsp;10<sup style='font-size: 65%;'>−3</sup></td></tr>
    <tr><td headers="Tumor" class="gt_row gt_left">CESC</td>
<td headers="Context" class="gt_row gt_left">SWB_NG_2015</td>
<td headers="Gene" class="gt_row gt_left">PIK3CA</td>
<td headers="p" class="gt_row gt_right">5.16&nbsp;×&nbsp;10<sup style='font-size: 65%;'>−4</sup></td>
<td headers="FDR" class="gt_row gt_right">8.77&nbsp;×&nbsp;10<sup style='font-size: 65%;'>−3</sup></td></tr>
    <tr><td headers="Tumor" class="gt_row gt_left">HNSC</td>
<td headers="Context" class="gt_row gt_left">ukbEUR_Height</td>
<td headers="Gene" class="gt_row gt_left">RHOA</td>
<td headers="p" class="gt_row gt_right">2.85&nbsp;×&nbsp;10<sup style='font-size: 65%;'>−4</sup></td>
<td headers="FDR" class="gt_row gt_right">9.40&nbsp;×&nbsp;10<sup style='font-size: 65%;'>−3</sup></td></tr>
    <tr><td headers="Tumor" class="gt_row gt_left">ESCA</td>
<td headers="Context" class="gt_row gt_left">IBD_NG2017.hap3</td>
<td headers="Gene" class="gt_row gt_left">SLC17A6</td>
<td headers="p" class="gt_row gt_right">1.38&nbsp;×&nbsp;10<sup style='font-size: 65%;'>−3</sup></td>
<td headers="FDR" class="gt_row gt_right">1.38&nbsp;×&nbsp;10<sup style='font-size: 65%;'>−2</sup></td></tr>
    <tr><td headers="Tumor" class="gt_row gt_left">KIRP</td>
<td headers="Context" class="gt_row gt_left">PD_GWAS_2017_N300k</td>
<td headers="Gene" class="gt_row gt_left">CUL3</td>
<td headers="p" class="gt_row gt_right">5.09&nbsp;×&nbsp;10<sup style='font-size: 65%;'>−3</sup></td>
<td headers="FDR" class="gt_row gt_right">2.03&nbsp;×&nbsp;10<sup style='font-size: 65%;'>−2</sup></td></tr>
  </tbody>
  
  
</table>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.2 (2024-10-31)
Platform: x86_64-conda-linux-gnu
Running under: Ubuntu 24.04.1 LTS

Matrix products: default
BLAS/LAPACK: /dartfs-hpc/rc/home/p/f0070pp/.conda/envs/diffdriver/lib/libopenblasp-r0.3.29.so;  LAPACK version 3.12.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: Etc/UTC
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] gt_0.11.1       dplyr_1.1.4     DT_0.33         workflowr_1.7.1

loaded via a namespace (and not attached):
 [1] Matrix_1.7-3      jsonlite_2.0.0    compiler_4.4.2    promises_1.3.2   
 [5] tidyselect_1.2.1  Rcpp_1.0.14       xml2_1.3.8        stringr_1.5.1    
 [9] git2r_0.35.0      callr_3.7.6       later_1.4.1       jquerylib_0.1.4  
[13] yaml_2.3.10       fastmap_1.2.0     lattice_0.22-6    R6_2.6.1         
[17] generics_0.1.3    knitr_1.50        htmlwidgets_1.6.4 tibble_3.2.1     
[21] rprojroot_2.0.4   bslib_0.9.0       pillar_1.10.1     diffdriver_0.1.6 
[25] rlang_1.1.5       utf8_1.2.4        cachem_1.1.0      stringi_1.8.7    
[29] httpuv_1.6.15     xfun_0.52         getPass_0.2-4     fs_1.6.5         
[33] sass_0.4.9        cli_3.6.4         withr_3.0.2       magrittr_2.0.3   
[37] crosstalk_1.2.1   ps_1.8.1          digest_0.6.37     grid_4.4.2       
[41] processx_3.8.6    rstudioapi_0.17.1 lifecycle_1.0.4   vctrs_0.6.5      
[45] data.table_1.17.0 evaluate_1.0.3    glue_1.8.0        whisker_0.4.1    
[49] rmarkdown_2.29    httr_1.4.7        tools_4.4.2       pkgconfig_2.0.3  
[53] htmltools_0.5.8.1</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
